src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,454 @@
+        return $.when<pagedResult<string>>({
+            items: this.allCategoryNames(),
+            totalResultCount: this.allCategoryNames().length,
+            resultEtag: null,
+            additionalResultInfo: undefined
+        })
+    }
+
+    private selectActionHandler(categoryToShow: string, event: JQueryEventObject) {
+        event.preventDefault();
+        this.setCategory(categoryToShow);
+    }    
+    
+    private setCategory(category: string) {
+        this.selectedCategory(category || this.allEntries()[0].data.Metadata.Category);
+        
+        if (this.categoriesGridController()) {
+            this.categoriesGridController().setSelectedItems([this.selectedCategory()]);
+        }
+    }
+
+    private getCategoryStateClass(category: string) {
+        let state;
+        let errorState;
+
+        this.allEntries().filter(entry => entry.data.Metadata.Category === category).forEach(entry => {
+                const entryState = this.getEntryStateClass(entry);
+                
+                if (entryState === "customized-item-with-error") {
+                    errorState = entryState;
+                }
+    
+                if (entryState === "customized-item") {
+                    state = entryState;
+                } 
+        });
+        
+        return errorState || state; 
+    }
+    
+    private getEntryStateClass(entry: models.settingsEntry) {
+        let stateClass;
+        
+        if (entry instanceof models.databaseEntry && entry.override()) {
+            stateClass = this.isValid(entry.validationGroup) ? "customized-item" : "customized-item-with-error";
+        }
+        
+        return stateClass;
+    }
+    
+    private initSummaryGrid() {
+        const summaryGrid = this.summaryGridController();
+        summaryGrid.headerVisible(true);
+
+        summaryGrid.init(() => this.fetchSummaryData(), () =>
+            [
+                new textColumn<models.settingsEntry>(summaryGrid, x => x.keyName(), "Configuration Key", "30%", {
+                    sortable: "string",
+                    extraClass: x => this.getEntryStateClass(x)
+                }),
+                new textColumn<models.settingsEntry>(summaryGrid, x => x.effectiveValue(), "Effective Value", "40%", {
+                    sortable: "string",
+                    extraClass: (x) => x.effectiveValue() ? "effective-value" : ""
+                }),
+                new textColumn<models.settingsEntry>(summaryGrid, x => x.effectiveValueOrigin(), "Origin", "20%", {
+                    sortable: "string",
+                    extraClass: (x) => {
+                        let classes = "source-item ";
+                        switch (x.effectiveValueOrigin()) {
+                            case "Database":
+                                classes += "source-item-database";
+                                return x.effectiveValue() ? classes : classes += " null-value";
```

Comment:
move this condition up  - to avoid repeating ourselves

Reply:
done